server/database/handle-goal-data.js

```diff
@@ -42,7 +65,14 @@ module.exports = (goal, user_id, callback) => {
     if (getGoalResult.rows.length) {
       // goal already exists, check for new ratings
       // adapt this when we have editing and deleting functionality
-      updateGoal(getGoalResult.rows[0], goal, callback);
+      const oldGoal = getGoalResult.rows[0];
+      if (goal.deleted) {
+        deleteGoal(goal, callback);
+      } else if (goal.edited) {
+        editGoal(goal, callback);
+      } else {
+        updateGoal(oldGoal, goal, callback);
+      }
```

Comment:
I guess there's also the case that the user might have edited the goal AND added a rating. In which case we should probs also run updateGoal after editing...